consensus/blocknode.go

```diff
@@ -58,143 +41,103 @@ type blockNode struct {
 	delayedSiacoinOutputs map[SiacoinOutputID]SiacoinOutput
 }
 
-// CheckTarget returns true if the block id is lower than the target.
-func (b Block) CheckTarget(target Target) bool {
-	blockHash := b.ID()
-	return bytes.Compare(target[:], blockHash[:]) >= 0
+// childDepth returns the depth of a blockNode's child nodes. The depth is the
+// "sum" of the current depth and current target, where sum is defined as:
+//
+//     sum(x,y) := 1/(1/x + 1/y)
+func (bn *blockNode) childDepth() (depth Target) {
+	cumulativeDifficulty := new(big.Rat).Add(bn.target.Inverse(), bn.depth.Inverse())
+	return RatToTarget(new(big.Rat).Inv(cumulativeDifficulty))
 }
 
-// Int returns a Target as a big.Int.
-func (t Target) Int() *big.Int {
-	return new(big.Int).SetBytes(t[:])
+// heavierThan returns true if the blockNode is sufficiently heavier than
+// 'cmp', where "sufficient" is defined as:
+//
+//     (1/bn.depth) - (1/cmp.depth) > (1/cmp.target) * SurpassThreshold
+func (bn *blockNode) heavierThan(cmp *blockNode) bool {
```

Comment:
wrt to "It seems like a potential place to make mistakes," I wish there was a way to indicate that function `foo` is *only* intended to be a helper function of function `bar`, so that `foo` could not be called outside of `bar`.
This can already be done by defining `foo` inside of `bar`, but the result is much less clean (and probably less efficient as well, since IIRC the compiler has to assume that `foo` may close around variables in `bar`).